switch/libwebsockets/0001-Add-platform-Nintendo-Switch.patch

```diff
@@ -0,0 +1,1112 @@
++
++	return -1;
++}
++
++#if defined(LWS_WITH_MBEDTLS)
++int
++lws_plat_mbedtls_net_send(void *ctx, const uint8_t *buf, size_t len)
++{
++	int fd = ((mbedtls_net_context *) ctx)->fd;
++	int ret;
++
++	if (fd < 0)
++		return MBEDTLS_ERR_NET_INVALID_CONTEXT;
++
++	ret = write(fd, buf, len);
++	if (ret >= 0)
++		return ret;
++
++	if (errno == EAGAIN || errno == EWOULDBLOCK)
++		return MBEDTLS_ERR_SSL_WANT_WRITE;
++
++	if (errno == EPIPE || errno == ECONNRESET)
++		return MBEDTLS_ERR_NET_CONN_RESET;
++
++	if( errno == EINTR )
++		return MBEDTLS_ERR_SSL_WANT_WRITE;
++
++	return MBEDTLS_ERR_NET_SEND_FAILED;
++}
++
++int
++lws_plat_mbedtls_net_recv(void *ctx, unsigned char *buf, size_t len)
++{
++	int fd = ((mbedtls_net_context *) ctx)->fd;
++	int ret;
++
++	if (fd < 0)
++		return MBEDTLS_ERR_NET_INVALID_CONTEXT;
++
++	ret = (int)read(fd, buf, len);
++	if (ret >= 0)
++		return ret;
++
++	if (errno == EAGAIN || errno == EWOULDBLOCK)
++		return MBEDTLS_ERR_SSL_WANT_READ;
++
++	if (errno == EPIPE || errno == ECONNRESET)
++		return MBEDTLS_ERR_NET_CONN_RESET;
++
++	if (errno == EINTR || !errno)
++		return MBEDTLS_ERR_SSL_WANT_READ;
++
++	return MBEDTLS_ERR_NET_RECV_FAILED;
++}
++#endif
+\ No newline at end of file
+diff --git a/lib/plat/nx/private-lib-plat-nx.h b/lib/plat/nx/private-lib-plat-nx.h
+new file mode 100644
+index 00000000..fcd633d4
+--- /dev/null
++++ b/lib/plat/nx/private-lib-plat-nx.h
+@@ -0,0 +1,68 @@
++/*
++* libwebsockets - small server side websockets and web server implementation
++*
++* Copyright (C) 2010 - 2019 Andy Green <andy@warmcat.com>
++*
++* Permission is hereby granted, free of charge, to any person obtaining a copy
++* of this software and associated documentation files (the "Software"), to
++* deal in the Software without restriction, including without limitation the
++* rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
++* sell copies of the Software, and to permit persons to whom the Software is
++* furnished to do so, subject to the following conditions:
++*
++* The above copyright notice and this permission notice shall be included in
++* all copies or substantial portions of the Software.
++*
++* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
++* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
++* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
++* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
++* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
++* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
++* IN THE SOFTWARE.
++*
++* Included from lib/private-lib-core.h if LWS_WITH_NX
++*/
++
++#include <unistd.h>
++#include <sys/types.h>
++#include <netdb.h>
++#include <netinet/in.h>
++#include <arpa/inet.h>
++#include <fcntl.h>
++#include <switch/kernel/mutex.h>
++#include <switch/services/bsd.h>
```

Comment:
don't use this bsd header. We translate between newlib fds & horizon fds internally. Use standard <sys/socket.h>